starport/cmd/network_chain_init.go

```diff
@@ -0,0 +1,193 @@
+package starportcmd
+
+import (
+	"fmt"
+	"os"
+	"strconv"
+	"sync"
+
+	"github.com/manifoldco/promptui"
+	"github.com/spf13/cobra"
+	"github.com/tendermint/starport/starport/pkg/cliquiz"
+	"github.com/tendermint/starport/starport/pkg/clispinner"
+	"github.com/tendermint/starport/starport/pkg/events"
+	"github.com/tendermint/starport/starport/services/chain"
+	"github.com/tendermint/starport/starport/services/network"
+)
+
+const (
+	flagRecover  = "recover"
+	flagMnemonic = "mnemonic"
+	flagKeyName  = "key-name"
+)
+
+// NewNetworkChainInit returns a new command to initialize a chain from a published chain ID
+func NewNetworkChainInit() *cobra.Command {
+	c := &cobra.Command{
+		Use:   "init [launch-id]",
+		Short: "Initialize a chain from a published chain ID",
+		Args:  cobra.ExactArgs(1),
+		RunE:  networkChainInitHandler,
+	}
+
+	c.Flags().Bool(flagRecover, false, "Recover chain account from a mnemonic")
+	c.Flags().String(flagMnemonic, "", "Mnemonic for recovered account")
+	c.Flags().String(flagKeyName, "", "key name for the chain account")
+
+	c.Flags().AddFlagSet(flagNetworkFrom())
+	c.Flags().AddFlagSet(flagSetHome())
+	c.Flags().AddFlagSet(flagSetKeyringBackend())
+	c.Flags().AddFlagSet(flagSetYes())
+
+	return c
+}
+
+func networkChainInitHandler(cmd *cobra.Command, args []string) error {
+	var (
+		recover, _  = cmd.Flags().GetBool(flagRecover)
+		mnemonic, _ = cmd.Flags().GetString(flagMnemonic)
+		keyName, _  = cmd.Flags().GetString(flagKeyName)
+	)
+
+	s := clispinner.New()
```

Comment:
Instead of duplicating the logic related to initializing the spinner, event bus, network.New and also checking if the home dir exists and overwriting it, we can move them under a common func to use both in the publish cmd and this one.

Reply:
> checking if the home dir exists
This is not the same home being check, `publish` doesn't check `spn/{launchID}` since launch ID doesn't exist yet